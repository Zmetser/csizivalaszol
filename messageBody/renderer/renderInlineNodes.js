/**
 * @flow
 */

import React from 'react'

import type { InlineNode } from '../types'

import {
  Text,
  Link
} from '../components'

function createInlineNode (node: InlineNode, index): ?React$Node<*> {
  switch (node.type) {
    case 'Text': return Text(node, index)
    case 'Link': return <Link {...node} key={index} />
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>) {
  return nodes.map(createInlineNode)
}
