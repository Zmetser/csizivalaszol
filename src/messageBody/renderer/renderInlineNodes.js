/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link,
  LineBreak
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return <Text node={node} key={index} />
    case 'Link': return <Link node={node} key={index} />
    case 'LineBreak': return <LineBreak key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
