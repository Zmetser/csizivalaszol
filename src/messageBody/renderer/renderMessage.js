import * as React from 'react'
import {
  Paragraph,
  List
} from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<*>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return <Paragraph node={node} key={index} />
      case 'List': return <List node={node} key={index} />
      default: throw new UnknownNodeError(node)
    }
  })
}
