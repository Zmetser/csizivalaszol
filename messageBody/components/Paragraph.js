/**
 * @flow
 */

import React from 'react'
import type { InlineNode } from '../types'

import renderInlineNodes from '../renderer/renderInlineNodes'

type Props = {
  value: Array<InlineNode>
}

export default function Paragraph ({ value }: Props) {
  return <p>{renderInlineNodes(value)}</p>
}
