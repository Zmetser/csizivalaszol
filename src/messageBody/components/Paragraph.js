import * as React from 'react'
import type { ParagraphNode } from '../types'

import renderInlineNodes from '../renderer/renderInlineNodes'

type Props = { node: ParagraphNode }
export default function Paragraph ({ node }: Props): React.Element<'p'> {
  return <p>{renderInlineNodes(node.value)}</p>
}
