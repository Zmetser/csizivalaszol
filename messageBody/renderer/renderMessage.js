/**
 * @flow
 */

import * as React from 'react'
import { Paragraph } from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<*>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return Paragraph({ value: node.value }, index)
      default: throw new UnknownNodeError(node)
    }
  })
}
