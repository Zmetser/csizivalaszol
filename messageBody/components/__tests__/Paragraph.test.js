import renderer from 'react-test-renderer'

import {
  createTextNode,
  createLink
} from '../../nodes.js'

import Paragraph from '../Paragraph'

const fooText = createTextNode('Foo', [])
const fooLink = createLink([fooText], '#')

describe('<Paragraph />', () => {
  test('with text', () => {
    const element = renderer.create(
      Paragraph({ value: [fooText] })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('with text, link and text', () => {
    const element = renderer.create(
      Paragraph({ value: [fooText, fooLink, fooText] })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })

  test('with styled text', () => {
    const element = renderer.create(
      Paragraph({ value: [
        createTextNode('Bold', ['Bold']),
        createTextNode('Italic', ['Italic']),
        createTextNode('Small and Code', ['Small', 'Code'])
      ] })
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
})
