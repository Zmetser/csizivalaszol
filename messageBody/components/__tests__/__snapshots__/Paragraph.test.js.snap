// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> with styled text 1`] = `
<p>
  <strong>
    Bold
  </strong>
  <em>
    Italic
  </em>
  <small>
    <code>
      Small and Code
    </code>
  </small>
</p>
`;

exports[`<Paragraph /> with text 1`] = `
<p>
  Foo
</p>
`;

exports[`<Paragraph /> with text, link and text 1`] = `
<p>
  Foo
  <a
    href="#"
  >
    Foo
  </a>
  Foo
</p>
`;
